cmd/namespace_registry_test.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/spf13/viper"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServeNamespaceRegistry(t *testing.T) {
+	issuerTempDir := t.TempDir()
+
+	viper.Reset()
+	config.InitConfig()
+	ikey := filepath.Join(issuerTempDir, "issuer.jwk")
+	viper.Set("IssuerKey", ikey)
+	viper.Set("Registry.DbLocation", filepath.Join(issuerTempDir, "test.sql"))
+	err := config.InitServer()
+	require.NoError(t, err)
+
+	err = nsregistry.InitializeDB()
+	require.NoError(t, err)
+	defer nsregistry.ShutdownDB()
+
+	gin.SetMode(gin.TestMode)
+	engine := gin.Default()
+
+	_, err = config.LoadPublicKey("", ikey)
+	require.NoError(t, err)
+	privKey, err := config.GetOriginJWK()
+	require.NoError(t, err)
+
+	//Configure registry
+	nsregistry.RegisterNamespaceRegistry(engine.Group("/"))
+
+	//Set up a server to use for testing
+	svr := httptest.NewServer(engine)
+	defer svr.CloseClientConnections()
+	defer svr.Close()
+
+	viper.Set("Federation.NamespaceUrl", svr.URL)
+	viper.Set("Origin.NamespacePrefix", "/test")
+
+	//Test functionality of registering a namespace (without identity)
+	err = nsregistry.NamespaceRegister(privKey, svr.URL+"/api/v1.0/registry", "", "/test")
+	require.NoError(t, err)
+
+	//Test we can list the namespace without an error
+	t.Run("Test namespace list", func(t *testing.T) {
+		//Set up a buffer to capture stdout
+		var stdoutCapture string
+		oldStdout := os.Stdout
+		r, w, _ := os.Pipe()
+		os.Stdout = w
+
+		//List the namespaces
+		err = nsregistry.NamespaceList(svr.URL + "/api/v1.0/registry")
+		require.NoError(t, err)
+		w.Close()
+		os.Stdout = oldStdout
+
+		capturedOutput := make([]byte, 1024)
+		n, _ := r.Read(capturedOutput)
+		stdoutCapture = string(capturedOutput[:n])
+		assert.Contains(t, stdoutCapture, `"Prefix":"/test"`)
+	})
+
+	//Test functionality of namespace get
+	t.Run("Test namespace get", func(t *testing.T) {
+		//Set up a buffer to capture stdout
+		var stdoutCapture string
+		oldStdout := os.Stdout
+		r, w, _ := os.Pipe()
+		os.Stdout = w
+
+		err = nsregistry.NamespaceGet(svr.URL + "/api/v1.0/registry")
+		require.NoError(t, err)
+		w.Close()
+		os.Stdout = oldStdout
+
+		capturedOutput := make([]byte, 1024)
+		n, _ := r.Read(capturedOutput)
+		stdoutCapture = string(capturedOutput[:n])
+		assert.Contains(t, stdoutCapture, `"Prefix":"/test"`)
+	})
+
+	t.Run("Test namespace delete", func(t *testing.T) {
+		//Test functionality of namespace delete
+		err = nsregistry.NamespaceDelete(svr.URL+"/api/v1.0/registry/test", "/test")
+		require.NoError(t, err)
+		var stdoutCapture string
+		oldStdout := os.Stdout
+		r, w, _ := os.Pipe()
+		os.Stdout = w
+		err = nsregistry.NamespaceGet(svr.URL + "/api/v1.0/registry")
+		require.NoError(t, err)
+		w.Close()
+		os.Stdout = oldStdout
+
+		capturedOutput := make([]byte, 1024)
+		n, _ := r.Read(capturedOutput)
+		stdoutCapture = string(capturedOutput[:n])
+		assert.Equal(t, "[]\n", stdoutCapture)
+	})
```

Comment:
Can you bookend your code with a `viper.Reset()`? If the next test to run happens not to call `viper.Reset()` at the start, it might leak some of your viper config.